code/GridFieldLimitItems.php

```diff
@@ -85,6 +121,20 @@ public function setRemoveFromTop($removeFromTop) {
         return $this;
     }
 
+    /**
+     * allows you to enforce the set limit by removing options to add new items
+     */
+    public function enforceLimit()
+    {
+        // ensure we are removing the buttons if these are
+        $this->onAfterManipulate(function(GridField $grid, SS_List $list){
+            if ($list->count() == $this->getMaxItems()) {
+                $grid->getConfig()->removeComponentsByType(GridFieldAddNewButton::class);
+                $grid->getConfig()->removeComponentsByType(GridFieldAddExistingAutocompleter::class);
+            }
+        });
+    }
+
```

Comment:
I believe there's a separate pull request to address this feature. For cleanliness and organization, and since this is open source, could you update this particular PR branch to not include the `enforceLimit` feature? Particularly since there could be divergence between the two PR's (haven't checked yet).

Reply:
PR #8